CMakeLists.txt

```diff
@@ -254,6 +254,39 @@ set(SDL_CFLAGS "")
 # avoid the DLL having a "lib" prefix
 if(WINDOWS)
   set(CMAKE_SHARED_LIBRARY_PREFIX "")
+  if(MSVC_VERSION GREATER 1910)
```

Comment:
I'm not sure that we want an MSVC check here. The compiler has nothing to do with whether an SDK header is available (and indeed MinGW has sdkddkver.h too).

I know this came from the existing code, so this is mostly a question for @slouken who wrote the original check in SDL_config_windows.h. Do you remember the reason for that that `_MSC_VER` check?

Reply:
I was targeting Visual Studio at the time and I didn't have other environments available for testing. As long as you guys have tested, I'm fine with whatever makes sense here.

Reply:
> 
> 
> I'm not sure that we want an MSVC check here. The compiler has nothing to do with whether an SDK header is available (and indeed MinGW has sdkddkver.h too). Right now we're just breaking WGI for Clang/GCC builds for seemingly no reason.
> 
> I know this came from the existing code, so this is mostly a question for @slouken who wrote the original check in SDL_config_windows.h. Do you remember the reason for that that `_MSC_VER` check?

We could drop the SDL_WINDOWS10_SDK define altogether in `SDL_rawinputjoystic.c`, and instead define it when `SDL_JOYSTICK_WGI` is also defined? 

That way, that whole thing only depends on the test for the `windows.gaming.input.h`. Would that work for the clang/mingw builds?

Reply:
(because in that case, we don't really care about "detecting the windows 10 SDK" here, and the patch would be a one-liner at the top of the rawinput driver)

Reply:
That would avoid having to do the CMake change, but we still need a way to enable the D3D11 renderer and WGI joystick driver in `SDL_config_windows.h` without CMake. Maybe just:
```
#include <sdkddkver.h>
#if _WIN32_WINNT >= 0x0601
#define SDL_VIDEO_RENDER_D3D11 1
#endif
#if _WIN32_WINNT >= 0x0A00
#define SDL_JOYSTICK_WGI 1
#endif
```

Do we think there's a scenario where somebody might want WGI for trigger rumble but not the whole WGI joystick driver?

Reply:
All I know is that, as it is currently setup, it works in builds made with VisualC\SDL.sln, but not with CMake, because the windows 10 SDK macro is never being enabled there. 

With the last change made in #4898, we now enable `SDL_JOYSTICK_WGI`, but the *actual* problem I have here is that `SDL_JOYSTICK_RAWINPUT_WGI` is still not enabled (see #4900)

Reply:
(the patch in question would look this this https://github.com/Ybalrid/SDL/commit/3b4a3be21b806e58aafb93d8a79fb85de4fc6b1f)